### ISSUE TITLE
"PLUG_VERSION_HEX" not displaying plugin version correctly in DAW Plugin Managers

### ISSUE THREAD

**Issue by rexmartin158** [iPlug2/iPlug2#832]
**Describe the bug**
DAWs do not display the correct plugin version number for VST2 plugins in their plugin managers (Cubase, Studio One, Tracktion, etc.)

**To Reproduce**
Check VST2 plugin version in DAW plugin manager.  Version shown for VST2 is not what is defined in config.h "#define PLUG_VERSION_HEX"

For example, #define PLUG_VERSION_HEX 0x00010203 shows up in DAW plugin managers as "1.0.2.0" (should be "1.2.3.0")

**IMPORTANT DETAILS**
* What plug-in format does it relate to - VST2
* What platform does it relate to - ALL
* What IGRAPHICS_BACKEND does it relate to: N/A

**Additional context**
Appears to be a carryover error from iPlug1 - did not work properly in iPlug1 either

See discussion and more info here: [https://iplug2.discourse.group/t/plug-version-hex-not-interpreted-correctly-in-daws/563](url)

**Comment by rexmartin158** [comment]
Fix proposed by TBProAudio in IPlugUtilities.h:

```
static inline int GetDecimalVersion(int versionInteger)
{
  int maj, min, pat;
  GetVersionParts(versionInteger, maj, min, pat);
 // return 10000 * maj + 100 * min + pat;
  return 1000 * maj + 100 * min + 10 * pat;//   TBPro fix for VST2 version numbers 2/5/2022
}
```

I have confirmed that this fixes the version number displayed for VST2 plugins in Cubase, Studio One, etc. 

**Need to make sure this change doesn't affect plugin versioning used in serialized data.

**Comment by AlexHarker** [comment]
This isn't involved in serialised data. It's just version reporting - it goes into FXP/FXB banks, however. I'm not sure how important that is.

However, I also want to check. The VS2 SDK suggests what is posted above would be correct, but in your forum post you say:

> #define PLUG_VERSION_HEX 0x00010203
> 
> ..... the VST2 plugin version shows up as “1.0.2.0”.

As far as I can see I would expect that to display as 10.2.0.3 with the current code.

**Comment by rexmartin158** [comment]
> As far as I can see I would expect that to display as 10.2.0.3 with the current code.

Yes, I would think that too but all the DAWs I tested showed "1.0.2.0".  I assume the reason is that DAWs read these numbers in hex.  So shifting "Version" or "Pat" to the wrong 10's position corrupted the numbers rather than just producing a decimal position shift.

Whatever the case, I have checked @TBProAudio 's suggested fix in Cubase, Studio One and Tracktion's Plugin Validator and it now shows up correct.

For example, this:  

```
#define PLUG_VERSION_HEX 0x00020304
#define PLUG_VERSION_STR "2.3.4"
```

Shows up in Cubase as:
![VST3](https://user-images.githubusercontent.com/84396036/152663976-7cada84c-87fd-43b3-af4d-0b0856a56cf5.png)
![VST2](https://user-images.githubusercontent.com/84396036/152663977-bfbdef12-36a9-4bed-a2ab-519282a70423.png)

And in Studio One as:
![Studio One](https://user-images.githubusercontent.com/84396036/152663978-dbeb12f1-d125-4e80-9fd3-09328e724d53.png)

In Tracktion's Pluginval as:
![Val](https://user-images.githubusercontent.com/84396036/152663985-54d08948-12a9-4599-a8d8-7710e59a8192.png)

All correct.

**Comment by AlexHarker** [comment]
That's great, but would you mind just indulging me and removing the fix with the same hex string above and confirming what it shows up as in this situation?

**Comment by rexmartin158** [comment]
Sorry, @AlexHarker , additional test today shows this fix **_almost_** works.  Please see below.

Original code:

```
static inline int GetDecimalVersion(int versionInteger)
{
  int maj, min, pat;
  GetVersionParts(versionInteger, maj, min, pat);
  return 10000 * maj + 100 * min + pat;
 // return 1000 * maj + 100 * min + 10 * pat;//   TBPro fix for VST2 version numbers 2/5/2022
}
```

with same hex string:
```
#define PLUG_VERSION_HEX 0x00020304
#define PLUG_VERSION_STR "2.3.4"
```

shows up in Cubase plugin manager as:
![Cubase VST3](https://user-images.githubusercontent.com/84396036/152844088-477550a3-67d3-428e-872f-2cfa20894259.png)
![Cubase VST2](https://user-images.githubusercontent.com/84396036/152844096-06dafae0-b772-48a6-9e64-b66d78bab8b0.png)

in Studio One as:
![studio one](https://user-images.githubusercontent.com/84396036/152844151-0de25c7d-6029-457c-98fd-a7996520c337.png)

and Pluginval as:
![pluginval](https://user-images.githubusercontent.com/84396036/152844208-eb32dd6b-ef55-4f87-a895-0c6e3d930dc1.png)

**Looks correct but - it's still not right!**

As I mentioned above, it's not just a simple 10's shift as it appears in @TBProAudio 's suggested fix.  It's all intertwined due to the hex numbers and also because VST2 shows 4 decimal digits and not 3.

As another test, I tried using the number 10 in the "min" position in both decimal and hex numbers:

```
#define PLUG_VERSION_HEX 0x00021004
#define PLUG_VERSION_STR "2.10.4"
```
and
```
#define PLUG_VERSION_HEX 0x00020A04
#define PLUG_VERSION_STR "2.10.4"
```

It should display as "2.10.4" but in both cases produced the same, wrong, result in Pluginval:
![pluginval2103](https://user-images.githubusercontent.com/84396036/152847777-d91962c6-dd97-4071-bfdf-d229a3da3419.png)

Again, it's the hex number causing problems here.  If we limit each section to only 0-9 it works OK with TB's fix.  If we go above 9 in any position then it starts adding/rolling over and messing up the next digit up.  Quite confusing!
